src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5868,12 +5868,12 @@ private ScanTypeArgumentListKind ScanTypeArgumentList(NameOptions options)
                         break;
 
                     case ScanTypeFlags.GenericTypeOrExpression:
-                        // See above.  If we have  X<Y<Z>,  then this would definitely be a type argument list.
-                        // However, if we have  X<Y<Z>> then this might not be type argument list.  This could just
-                        // be some sort of expression where we're comparing, and then shifting values.
-                        if (!isDefinitelyTypeArgumentList)
+                        // See above.  If we have `X<Y<Z>,` or `X<Y<Z>)`  then this would definitely be a type argument
+                        // list. However, if we have `X<Y<Z>>` then this might not be type argument list.  This could
+                        // just be some sort of expression where we're comparing, and then shifting values.
+                        isDefinitelyTypeArgumentList = isDefinitelyTypeArgumentList || this.CurrentToken.Kind is SyntaxKind.CommaToken or SyntaxKind.CloseParenToken;
```

Comment:
> or SyntaxKind.CloseParenToken [](http://example.com/codeflow?start=134&length=30)

Were we missing this case previously? Is there a test that hits this case?

Reply:
i did it for completeness.  i can add an explicit test for it as well.

Reply:
If this was updated for consistency with other callers, would it make sense to add back the `DetermineIfDefinitelyTypeArgumentList()` method? 

> i can add an explicit test for it as well.

Yes, please add a test if there is an observable difference, thanks.